currencies/rate_converter.go

```diff
@@ -93,6 +89,10 @@ func (rc *RateConverter) fetch() (*Rates, error) {
 		return nil, err
 	}
 
+	if response.StatusCode >= 400 {
```

Comment:
Yeah this addresses a bug I found. If the rate converter returned a non-2xx with a well-formed response body, no error would be caught so the request would be considered successful. As a result, if we had successfully retrieved the currency rates prior to the non-2xx's, we would never fall back to constant rates because we would never see the rates as stale.
The old tests checked for 4xx codes but all of the tests had nil response bodies which resulted in an error when either `ioutil.ReadAll` or `json.Unmarshal` was executed, I can't remember which.

Reply:
Should we not be checking just for == 200 then ?
The other 2xx and 3xx codes do return useful data either, and we would see a unmarshal error instead of the unexpected status code, which is less useful for troubleshooting.

Reply:
The 3xx status codes will be handled by the networking library. So it's really then just the 2xx codes which is a concern. I agree it would be strange to get a non-200 code from the rate converter service, but really any 2xx is considered success so I'm fine with accepting that as a valid server response and letting the marshal code below this handle any errors with the content of the response.